src/Init/Data/List/Lemmas.lean

```diff
@@ -1139,6 +1177,63 @@ theorem eq_replicate {a : α} {n} {l : List α} :
   ⟨fun h => h ▸ ⟨length_replicate .., fun _ => eq_of_mem_replicate⟩,
    fun ⟨e, al⟩ => e ▸ eq_replicate_of_mem al⟩
 
+@[simp] theorem append_replicate_replicate : replicate n a ++ replicate m a = replicate (n + m) a := by
```

Comment:
The general principle is that `Nat` arithmetic is cheap (`omega`!).

However there is definitely something very smelly here, as `replicate_succ` *is* a `simp` lemma. I am going to investigate how things look without that; but it will be bad!

Reply:
(e.g. a parallel is that we have `List.drop_drop` rather than the opposite direction `List.drop_add`. Although this is a recent change I made, so it is not exactly precedent!)

Reply:
It's easy to adjust the proofs so that we are agnostic as to whether `replicate_succ` is a `simp` lemma. I will think more on this one...

Reply:
Oh, removing `@[simp]` from `replicate_succ` solves a (very minor) confluence problem... I'm tempted.

Reply:
Yeah, confluence will suggest that if `replicate_succ` is simp, then you'd want `replicate_append` and not the other way around - or conversely that if we want this lemma we maybe want `cons_replicate` rewriting cons to succ. But then rewrite rules matching on constructor applications are smelly as well.